test/qa-tests/jstests/restore/users_and_roles_admin.js

```diff
@@ -0,0 +1,193 @@
+// Tests that mongodump and mongorestore properly handle access control information when doing
```

Comment:
This was a test in legacy30, but the stuff it was testing wasn't covered anywhere in the new jstests or integration tests. As such, I copied it here and updated parts of it.